eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs

```diff
@@ -206,16 +206,21 @@ deriving stock instance
   Show (UtxoPredicateFailure era)
 
 deriving stock instance
-  ( Eq (Core.Value era),
+  ( CC.Crypto (Crypto era),
+    Eq (Core.Value era),
     Eq (Core.TxOut era),
-    CC.Crypto (Crypto era),
     Eq (PredicateFailure (Core.EraRule "UTXOS" era))
   ) =>
   Eq (UtxoPredicateFailure era)
 
 instance
   ( Era era,
-    Shelley.TransUTxOState NoThunks era,
```

Comment:
I believe that the `TransFoo` classes were originally added to make constraints like this more compact. I see in the PR notes that you noticed this added superfluous constraints. It's a shame we can't find some middle ground...

Reply:
@JaredCorduan I don't think removing a few extra lines in a list of constraints are worth the trouble of adding a new concept (namely `Trans*`) and they are definitely not worth redundant constraints that the concept introduces. It takes almost no effort to add these constraints manually, moreover compiler tells us exactly which constraints are missing, which is why all theses changes in this PR where extremely trivial and easy for me to do. 

Also note that for example for functions that use `show` on `UTxOState`, it is enough to add `Show (UTxOState era)` instead of `TransUTxOState Show era` or listing all the constraints manually.